src/main/java/core/basesyntax/service/impl/ReturnOperation.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.RecordDao;
+import core.basesyntax.dao.RecordDaoImpl;
+import core.basesyntax.model.Product;
+import core.basesyntax.record.Record;
+import core.basesyntax.service.DataOperation;
+
+public class ReturnOperation implements DataOperation {
+    private final RecordDao recordDao = new RecordDaoImpl();
```

Comment:
If you already used a singleton pattern, I believe you should make RecordDao singleton as well, to avoid creating this object in each operation